packages/umbra_flutter/lib/src/umbra_exception.dart

```diff
@@ -0,0 +1,215 @@
+/// {@template umbra_exception}
+/// Exception thrown when an error occurs during shader compilation.
+/// {@endtemplate}
+abstract class UmbraException extends Error {
+  /// {@macro umbra_exception}
+  factory UmbraException(Exception exception) {
+    final stringifiedException = exception.toString();
+    final opCode = int.tryParse(stringifiedException.split(':')[0].trim());
+    if (opCode != null) {
+      final message = stringifiedException.split(':')[1].trim();
+      if (message == 'Not a supported op.') {
+        return UnsupportedOperator._(opCode);
+      }
+      throw Exception('Unknown transpiler exception message: $message');
+    }
+    throw Exception('Unsupported exception type: $stringifiedException');
```

Comment:
might want to use `Error.throwWithStackTrace` to preserve the original stackTrace.